### DISCUSSION TITLE
How do I get use the background/foreground/cursor color in yazi?

### DISCUSSION THREAD

**Discussion by UtkarshVerma** [sxyazi/yazi#2094] [1. Q&A] [answered]
### What system are you running Yazi on?

Linux X11

### What terminal are you running Yazi in?

st

### `yazi --debug` output

```Shell
Yazi
    Version: 0.4.1 (Arch Linux 2024-12-19)
    Debug  : false
    Triple : x86_64-unknown-linux-gnu (linux-x86_64)
    Rustc  : 1.83.0 (90b35a62 2024-11-26)

Ya
    Version: 0.4.1 (Arch Linux 2024-12-19)

Emulator
    Brand.from_env      : None
    Emulator.detect     : Emulator { kind: Right(Unknown { kgp: false, sixel: true }), light: false, cell_size: Some((11, 21)) }
    Emulator.detect_full: Ok(Emulator { kind: Right(Unknown { kgp: false, sixel: true }), light: false, cell_size: Some((11, 21)) })

Adapter
    Adapter.matches: Sixel

Desktop
    XDG_SESSION_TYPE           : Some("x11")
    WAYLAND_DISPLAY            : None
    DISPLAY                    : Some(":0")
    SWAYSOCK                   : None
    HYPRLAND_INSTANCE_SIGNATURE: None
    WAYFIRE_SOCKET             : None

SSH
    shared.in_ssh_connection: false

WSL
    WSL: false

NVIM
    NVIM          : false
    Neovim version: 0.10.3

Variables
    SHELL           : Some("/bin/zsh")
    EDITOR          : Some("nvim")
    VISUAL          : None
    YAZI_FILE_ONE   : None
    YAZI_CONFIG_HOME: None
    YAZI_ZOXIDE_OPTS: None
    FZF_DEFAULT_OPTS: None

Text Opener
    default     : Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block-create: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block-rename: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })

Multiplexers
    TMUX               : 0
    tmux version       : tmux 3.5a
    tmux build flags   : enable-sixel=Unknown
    ZELLIJ_SESSION_NAME: None
    Zellij version     : No such file or directory (os error 2)

Dependencies
    file          : 5.46
    ueberzugpp    : No such file or directory (os error 2)
    ffmpeg/ffprobe: 7.1 / 7.1
    pdftoppm      : 24.11.0
    magick        : 7.1.1-41
    fzf           : 0.57.0
    fd/fdfind     : 10.2.0 / No such file or directory (os error 2)
    rg            : 14.1.1
    chafa         : 1.14.5
    zoxide        : 0.9.6
    7z/7zz        : 24.09 / No such file or directory (os error 2)
    jq            : 1.7.1

Clipboard
    wl-copy/paste: No such file or directory (os error 2) / No such file or directory (os error 2)
    xclip        : 0.13
    xsel         : No such file or directory (os error 2)
```

### Describe the question

I know I can use the base 16 colors (color0-color15) in yazi's theme.toml. However, these colors do not include the following colors:
- background
- foreground
- cursor

We do have escape sequences to query them.

### Why do I need it?
I want to make a base16 flavor for yazi where I don't want to hardcode the background value.

So far, I got up to here but I can't think of a way to fix the separator.

![image](https://github.com/user-attachments/assets/324eea4b-dd54-4952-b2ef-2c5bcabd7b74)

```toml
[mode]
normal_main = { fg = "blue", bg = "reset", reversed = true, bold = true }
normal_alt = { fg = "blue", bg = "black" }
````

### Checklist

- [X] I have searched the existing discussions/issues

**Comment by sxyazi** [comment]
If you want to use the terminal default background, just do not set the `bg`, since it's already the default:

```diff
- normal_alt = { fg = "blue", bg = "black" }
+ normal_alt = { fg = "blue" }
```

`fg` is also the same

**Reply by UtkarshVerma** [reply]
`normal_alt` is fine in this case. What I want to address here is that `normal_main` should by styled like so:
- fg = terminal's background
- bg = blue

But when I do this, the separator starts behaving weirdly.

**Reply by sxyazi** [reply]
Why would you want to set the background as the foreground color? 

I don't think that's possible. OSC 11 responds to RGB colors, but you want base16 colors, and not all terminals support OSC 11.

**Reply by UtkarshVerma** [reply]
Basically, I want to reverse the foreground and background colors for the mode indicator, since that is how the UI should be.

![image](https://github.com/user-attachments/assets/4f621816-cb55-49f8-8c5c-285e518a4133)

In this case, everything works. Is there a way to ensure that the separator is styled correctly too? I want it to be blue, just like `normal_main`

**Reply by sxyazi** [reply, marked as answer]
> I want to reverse the foreground and background colors for the mode indicator

Set `reversed = true`, quote https://yazi-rs.github.io/docs/configuration/theme#types.style:

> reversed (Boolean): Reversed foreground and background colors
